src/main/java/de/medizininformatikinitiative/torch/service/FilterService.java

```diff
@@ -125,10 +125,10 @@ private static boolean isValidSearchParam(SearchParameter searchParam) {
                         searchParam.getType().equals(Enumerations.SearchParamType.DATE));
     }
 
-    private record CompiledFilter(List<ParsedFilter> filters,
+    public record CompiledFilter(List<ParsedFilter> filters,
```

Comment:
please do not make the records public only for tests. Please test only the public interface if FilterService.